unity-renderer/Assets/DCLServices/WearablesCatalogService/LambdasWearablesCatalogService.cs

```diff
@@ -0,0 +1,308 @@
+            {
+                thirdPartyCollectionPagePointers[(userId, collectionId)] = pagePointer = new (
+                    WEARABLES_BY_THIRD_PARTY_COLLECTION_END_POINT
+                       .Replace("{userId}", userId)
+                       .Replace("{collectionId}", collectionId),
+                    pageSize, ct, this);
+            }
+
+            var pageResponse = await pagePointer.GetPageAsync(pageNumber, ct);
+            AddWearablesToCatalog(pageResponse.response.wearables);
+
+            return pageResponse.response.wearables;
+        }
+
+        public async UniTask<WearableItem> RequestWearableAsync(string wearableId, CancellationToken ct)
+        {
+            if (WearablesCatalog.TryGetValue(wearableId, out WearableItem wearable))
+            {
+                if (wearablesInUseCounters.ContainsKey(wearableId))
+                    wearablesInUseCounters[wearableId]++;
+
+                return wearable;
+            }
+
+            ct.ThrowIfCancellationRequested();
+
+            try
+            {
+                // All the requests happened during the same frames interval are sent together
+                return await SyncWearablesRequestsAsync(wearableId, ct);
+            }
+            catch (OperationCanceledException) { return null; }
+        }
+
+        public void AddWearablesToCatalog(IEnumerable<WearableItem> wearableItems)
+        {
+            foreach (WearableItem wearableItem in wearableItems)
+            {
+                if (WearablesCatalog.ContainsKey(wearableItem.id))
+                    continue;
+
+                wearableItem.SanitizeHidesLists();
+                WearablesCatalog.Add(wearableItem.id, wearableItem);
+
+                if (!wearablesInUseCounters.ContainsKey(wearableItem.id))
+                    wearablesInUseCounters.Add(wearableItem.id, 1);
+            }
+        }
+
+        public void RemoveWearablesFromCatalog(IEnumerable<string> wearableIds)
+        {
+            foreach (string wearableId in wearableIds)
+            {
+                WearablesCatalog.Remove(wearableId);
+                wearablesInUseCounters.Remove(wearableId);
+            }
+        }
+
+        public void RemoveWearablesInUse(IEnumerable<string> wearablesInUseToRemove)
+        {
+            foreach (string wearableToRemove in wearablesInUseToRemove)
+            {
+                if (!wearablesInUseCounters.ContainsKey(wearableToRemove))
+                    continue;
+
+                wearablesInUseCounters[wearableToRemove]--;
+            }
+        }
+
+        public void EmbedWearables(IEnumerable<WearableItem> wearables)
+        {
+            foreach (WearableItem wearableItem in wearables)
+            {
+                WearablesCatalog[wearableItem.id] = wearableItem;
+
+                if (wearablesInUseCounters.ContainsKey(wearableItem.id))
+                    wearablesInUseCounters[wearableItem.id] = 10000; //A high value to ensure they are not removed
```

Comment:
Just set `int.MaxValue`

Reply:
Done in: https://github.com/decentraland/unity-renderer/pull/4084/commits/62738dd3619fccb4409f386166de51b1e0cec8af